slider/index.js

```diff
@@ -0,0 +1,170 @@
+class IRNMNSlider extends HTMLElement {
+
+    CLASSNAMES = [];
+
+    constructor() {
+        super();
+        this.CLASSNAMES = this.selectors;
+        this.currentSlide = 1; // Start from the first visible slide after cloning        
+    }
+
+    /**
+     * Get the selectors from the attribute
+     * to be used as classnames obj
+     * 
+     * @returns {object} - The selectors object
+     */
+    get selectors() {
+        let selectors = this.getAttribute('selectors');
+        selectors = JSON.parse(selectors);
+        let classnames = [];
+        // replace classnames with selectors
+        for (let key in selectors) {
+            classnames[key.toUpperCase()] = selectors[key];
+        }
+
+        return classnames;
+    }
+
+    connectedCallback() {
+        this.initSlider();
+    }
+
+    initSlider() {
+
+        const swipeContainer = this.querySelector(this.CLASSNAMES.SWIPE_CONTAINER);
+        if ( !swipeContainer ) {
+            console.error('Swipe container not found');
+            return;
+        }
+        const slides = swipeContainer.querySelectorAll(this.CLASSNAMES.SLIDES);
+        const totalSlides = slides.length;
+
+        if (this.dataset.sliderInitialized === "true") {
+            return;
+        }
+        this.dataset.sliderInitialized = "true";
+
+        /**
+         * Handle slider loopping
+         */
+        const firstClone = slides[0].cloneNode(true);
+        const lastClone = slides[slides.length - 1].cloneNode(true);
+        swipeContainer.appendChild(firstClone); // Add the first clone at the end
+        swipeContainer.insertBefore(lastClone, slides[0]); // Add the last clone at the beginning
+
+        // Update total slides including clones
+        const allSlides = swipeContainer.querySelectorAll(this.CLASSNAMES.SLIDES);
+        const clonedSlidesCount = allSlides.length;
+
+        this.querySelector(this.CLASSNAMES.TOTAL_SLIDES).textContent = totalSlides;
+
+
+        /**
+         * Update slides position using translateX method
+         * 
+         * @returns { void }
+         */
+        const updateSlides = () => {
+            swipeContainer.style.transition = 'transform 0.3s ease';
```

Comment:
@samuelsilvapt is there a way we can move the ease and timing values to a CSS custom property so they can be easily overwritten?

Reply:
I added an option to the transition be defined in the HTML props